tests/Builder/Accumulator/AccumulatorAccumulatorTest.php

```diff
@@ -0,0 +1,147 @@
+                        }
+                    }
+                }
+            }
+        ]
+        JSON;
+    }
+
+    /**
+     * THIS METHOD IS AUTO-GENERATED. ANY CHANGES WILL BE LOST!
+     *
+     * @see testUseInitArgsToVaryTheInitialStateByGroup
+     */
+    public function getExpectedUseInitArgsToVaryTheInitialStateByGroup(): string
+    {
+        return <<<'JSON'
+        [
+            {
+                "$group": {
+                    "_id": {
+                        "city": "$city"
+                    },
+                    "restaurants": {
+                        "$accumulator": {
+                            "init": {
+                                "$code": "function (city, userProfileCity) { return { max: city === userProfileCity ? 3 : 1, restaurants: [] } }"
+                            },
+                            "initArgs": [
+                                "$city",
+                                "Bettles"
+                            ],
+                            "accumulate": {
+                                "$code": "function (state, restaurantName) { if (state.restaurants.length < state.max) { state.restaurants.push(restaurantName); } return state; }"
+                            },
+                            "accumulateArgs": [
+                                "$name"
+                            ],
+                            "merge": {
+                                "$code": "function (state1, state2) { return { max: state1.max, restaurants: state1.restaurants.concat(state2.restaurants).slice(0, state1.max) } }"
+                            },
+                            "finalize": {
+                                "$code": "function (state) { return state.restaurants }"
+                            },
+                            "lang": "js"
+                        }
+                    }
+                }
+            }
+        ]
+        JSON;
+    }
+
+    /** @see getExpectedUseAccumulatorToImplementTheAvgOperator */
+    public function testUseAccumulatorToImplementTheAvgOperator(): void
+    {
+        $pipeline = new Pipeline(
+            Stage::group(
+                _id: Expression::fieldPath('author'),
+                avgCopies: Accumulator::accumulator(
+                    init: new Javascript('function () { return { count: 0, sum: 0 } }'),
+                    accumulate: new Javascript('function (state, numCopies) { return { count: state.count + 1, sum: state.sum + numCopies } }'),
+                    accumulateArgs: [Expression::fieldPath('copies')],
+                    merge: new Javascript('function (state1, state2) { return { count: state1.count + state2.count, sum: state1.sum + state2.sum } }'),
+                    finalize: new Javascript('function (state) { return (state.sum / state.count) }'),
+                    lang: 'js',
+                ),
+            ),
+        );
+
+        $expected = $this->getExpectedUseAccumulatorToImplementTheAvgOperator();
```

Comment:
If I understand correctly, only the `get` methods are generated and you still need to write the test yourself. Rather than embed generated methods _within_ non-generated _files_, why not just generate JSON files which are then referenced from these tests?

Alternatively, you can generate actual PHP files that return the value to compare against. I suppose that could be either an array or `MongoDB\BSON\PackedArray` instance.